File: OLD SNACKS/PYTHON/phone_book.py
contacts = []

def add_contact(name, last_name, phone_number):
    contact = {'name': name, 'last_name': last_name, 'phone_number': phone_number}
    contacts.append(contact)
    print(f"Contact {name} {last_name} added.")

def remove_contact(phone_number):
    for contact in contacts:
        if contact['phone_number'] == phone_number:
            contacts.remove(contact)
            print(f"Contact with phone number {phone_number} removed.")
            return
    print("Contact not found.")

def contact_found_by_phone_number(phone_number):
    for contact in contacts:
        if contact['phone_number'] == phone_number:
            print(f"Contact Found: {contact['name']} {contact['last_name']}, Phone: {contact['phone_number']}")
            return
    print("Contact not found.")

def contact_found_by_first_name(name):
    found = False
    for contact in contacts:
        if contact['name'] == name:
            print(f"Contact Found: {contact['last_name']} {contact['phone_number']}, First Name: {contact['name']}")
            found = True
    if not found:
        print("Contact not found with first name.")

def contact_found_by_last_name(last_name):
    found = False
    for contact in contacts:
        if contact['last_name'] == last_name:
            print(f"Contact Found: {contact['name']} {contact['phone_number']}, Last Name: {contact['last_name']}")
            found = True
    if not found:
        print("Contact not found with last name.")

def edit_contact(phone_number):
    for contact in contacts:
        if contact['phone_number'] == phone_number:
            print(f"Editing Contact: {contact['name']} {contact['last_name']}, Phone: {contact['phone_number']}")
            new_name = input("Enter new First Name:   ")
            new_last_name = input("Enter new Last Name:   ")
            new_phone_number = input("Enter new Phone Number:   ")

            # Update fields if new values are provided
            if new_name:
                contact['name'] = new_name
            if new_last_name:
                contact['last_name'] = new_last_name
            if new_phone_number:
                contact['phone_number'] = new_phone_number
            
            print("Contact updated successfully.")
            return
    print("Contact not found.")

def main():
    while True:
        menu_options = """
Welcome to PhoneBook 

Choose An Option
1 => Add Contact
2 => Remove Contact
3 => Find Contact By Phone Number
4 => Find Contact By First Name
5 => Find Contact By Last Name 
6 => Edit Contact
0 => Exit
"""
        print(menu_options)
        try: 
            user_input = int(input("Enter Your Choice: "))
            match user_input:
                case 1: 
                    name = input("Enter First Name: ")
                    last_name = input("Enter Last Name: ")
                    phone_number = input("Enter Phone Number: ")
                    add_contact(name, last_name, phone_number)
                case 2: 
                    phone_number = input("Enter Phone Number to Remove: ")
                    remove_contact(phone_number)
                case 3:
                    phone_number = input("Enter Phone Number to Find Contact: ")
                    contact_found_by_phone_number(phone_number)
                case 4: 
                    name = input("Enter First Name to Find Contact: ")
                    contact_found_by_first_name(name)
                case 5:
                    last_name = input("Enter Last Name to Find Contact: ")
                    contact_found_by_last_name(last_name)
                case 6:
                    phone_number = input("Enter Phone Number to Edit Contact: ")
                    edit_contact(phone_number)
                case 0:
                    print("Exiting PhoneBook. Goodbye!")
                    break
                case _:
                    print("Invalid option. Please try again.")
        except ValueError:
            print("Invalid input. Please enter a number.")

if __name__ == "__main__":
    main()

File: OLD SNACKS/JAVA/PhoneBook.java
import java.util.ArrayList;
import java.util.Scanner;

public class PhoneBook{
	private static ArrayList<String[]> contacts = new ArrayList<>();

	public static void addContact(String name, String lastName, String phoneNumber){
		contacts.add(new String[]{name, lastName, phoneNumber});
		System.out.println("Contact " + name + " " + lastName + " added.");
	}
	
	public static void removeContact(String phoneNumber){
		boolean found = false;
		for (int i = 0; i < contacts.size(); i++){
			if(contacts.get(i)[2].equals(phoneNumber)) {
				contacts.remove(i);
				System.out.println("Contact with Phone number " + phoneNumber + " removed");
				found = true;                                                                                                                                                                                nb                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
				break;
			}
		}
		if(!found){
			System.out.println("Contact not found");
		}
	}

	public static void findContactByPhoneNumber(String phoneNumber){
		for (String[] contact : contacts){
			if(contact[2].equals(phoneNumber)){
				System.out.println("Contact Found: " + contact[0] + "  " + contact[1] + "  " + "phoneNumber:  " + contact[2]);
				return;
			} 
		}
		System.out.println("Contact not Found");
	}

	public static void findContactByFirstName(String name){
		boolean found = false;
		for (String[] contact : contacts){
			if(contact[0].equals(name)){
				System.out.println("Contact Found: " + contact[1] + "  " + contact[2] + "  " + " Name:   " + contact[0]);
				found = true;
			} 
		}
		if(!found){
			System.out.println("Contact not found");
		}
	}

	public static void findContactByLastName(String lastName){
		boolean found = false;
		for (String[] contact : contacts){
			if(contact[1].equals(lastName)){
				System.out.println("Contact Found: " + contact[0] + "  " + contact[2] + "  " + " Last Name:   " + contact[1]);
				found = true;
			} 
		}
		if(!found){
			System.out.println("Contact not Found.");
		}
	}
	
	public static void editContact(String phoneNumber){
		for(String[] contact : contacts){
			if(contact[2].equals(phoneNumber)){
				Scanner input = new Scanner(System.in);
				System.out.println("Enter new First Name:   ");
				String newName = input.nextLine();
				System.out.println("Enter new Last Name:   ");
				String newLastName = input.nextLine();
				System.out.println("Enter new Phone Number:   ");
				String newPhoneNumber = input.nextLine();
				

				if (!newName.isEmpty()){
					contact[0] = newName;
				} 
				if (!newLastName.isEmpty()){
					contact[1] = newLastName;
				}
				if (!newPhoneNumber.isEmpty()){
					contact[2] = newPhoneNumber;
				}
				System.out.println("Contact Update Successfully.");
				return;
			}
		}
		System.out.println("Contact not Found.");
	}

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);

		int userInput = -1;
		
		System.out.println("Welcome to Phone Book App: ");
		while(userInput != 1) {
			System.out.println("""
					Choose an Option
					1 => Add Contact
					2 => Remove Contact
					3 => Find Contact By Phone Number
					4 => Find Contact By First Name 
					5 => Find Contact By Last Name
					6 => Edit Contact	
					0 => Edit Contact
					""");
			System.out.println("Enter your Choice:   ");
			userInput = input.nextInt();
			input.nextLine();

			switch(userInput){
				case 1 -> {
					System.out.println("Enter First Name:   ");
					String name = input.nextLine();
					System.out.println("Enter Last Name:   ");
					String lastName = input.nextLine();
					System.out.println("Enter Phone Number:   ");
					String phoneNumber = input.nextLine();
					addContact(name, lastName, phoneNumber);
				}
				case 2 -> {
					System.out.println("Enter Phone Number to Remove:   ");
					String phoneNumber = input.nextLine();
					removeContact(phoneNumber);
				}
				case 3 -> {
					System.out.println("Enter Phone Number to Find Contact:   ");
					String phoneNumber = input.nextLine();
					findContactByPhoneNumber(phoneNumber);
				}
				case 4 -> {
					System.out.println("Enter First Name to Find Contact:   ");
					String name = input.nextLine();
					findContactByFirstName(name);
				}
				case 5 -> {
					System.out.println("Enter Last Name to Find Contact:   ");
					String lastName = input.nextLine();
					findContactByLastName(lastName);
				}
				case 6 -> {
					System.out.println("Enter Phone Number to Edit Contact:   ");
					String phoneNumber = input.nextLine();
					editContact(phoneNumber);
				}
				case 0 -> {
					System.out.println("Exiting.....");
				}
				default -> System.out.println("Invalid Choice. Please Try agian.");
			}
		}
	}
}